ipset.go

```diff
@@ -332,6 +333,77 @@ type IPSet struct {
 	rr []IPRange
 }
 
+// ParseIPSet parses a comma-separated list of IP ranges into an IPSet.
+// If an invalid IPRange is provided, an empty IPSet and an error are returned.
+// The provided list need not be representative of a true set; ranges may
+// overlap and duplicates are allowed.
+// An empty string produces an empty set.
+func ParseIPSet(s string) (*IPSet, error) {
+	var (
+		b   IPSetBuilder
+		r   IPRange
+		err error
+	)
+	if len(s) == 0 {
+		return b.IPSet()
+	}
+	rss := strings.Split(s, ",")
+	for _, rs := range rss {
+		r, err = ParseIPRange(rs)
+		if err != nil {
+			return &IPSet{}, fmt.Errorf("invalid IP range %q in set %q", rs, s)
+		}
+		b.AddRange(r)
+	}
+	return b.IPSet()
+}
+
+// MustParseIPSet calls ParseIPSet(s) and panics if an error is encountered.
+func MustParseIPSet(s string) *IPSet {
+	ret, err := ParseIPSet(s)
+	if err != nil {
+		panic(err)
+	}
+	return ret
+}
+
+// String returns a string representation of the IPSet.
+func (s IPSet) String() string {
+	var b strings.Builder
+	for i, r := range s.rr {
+		b.WriteString(r.String())
```

Comment:
Great catch.

Reply:
Bypassed the use of the `strings.Builder` altogether, since it's just using a `[]byte` internally anyway.

Reply:
strings.Builder will probably eliminate an allocation, because the internal `[]byte` can be unsafely re-used as the resulting string. If you're curious, add a benchmark?

Reply:
Hey neat, you were absolutely right about the extra allocation. Sometimes I forget that stdlib doesn't stray away from `unsafe`. Thanks for the insight!